File: platforms/windows/src/shared/easter-egg.js
(function exposeEasterEgg(global) {
  function daysTogether(now = new Date()) {
    const base = Date.UTC(2024, 0, 30);
    const today = Date.UTC(now.getFullYear(), now.getMonth(), now.getDate());
    return Math.floor((today - base) / 86400000);
  }

  function message(now = new Date()) {
    return `谨以此app，纪念Eric与Eva认识${daysTogether(now)}天！`;
  }

  const api = { daysTogether, message };
  if (typeof module !== "undefined" && module.exports) {
    module.exports = api;
  }
  global.NCMEasterEgg = api;
})(typeof window !== "undefined" ? window : globalThis);

File: apps/windows/NcmBatchMp3.Core/EasterEgg.cs
namespace NcmBatchMp3.Core;

public static class EasterEgg
{
    private static readonly DateTime BaseDate = new(2024, 1, 30);

    public static int DaysTogether(DateTime now)
    {
        return (now.Date - BaseDate).Days;
    }

    public static string Message(DateTime now)
    {
        return $"谨以此app，纪念Eric与Eva认识{DaysTogether(now)}天！";
    }
}
